### PR TITLE
Fix: LDLf Atomic symbols are now converible formulas.

### PR DESCRIPTION
**Pull request by cipollone**
## Proposed changes

From commit message:
-  Parsing: outside a regular expression, now everything is parsed as a
    LDLfFormula. PL symbols only allowed in RegExp.
- Checked LDLfPropositionalAtom, written LDLfPropositionalTrue and
    LDLfPropositionalFalse.
- Not using FiniteTraceWrapper anymore.

Now, anything expressible with the parser is a composition of `LDLfFormula`. This solves a lot of problems. Solves #25 .
(FiniteTraceWrapper is no nonger necessary).

## Fixes

Solves #25 and (i think) many more issues.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../master/CONTRIBUTING.rst) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by cipollone** [comment]
Meanwhile I've also tried to set `LDLfFormula.to_nnf` as abstract, but it gave me an error on `LDLfAnd` and `LDLfOr`. I hope this is fine.

**Comment by cipollone** [comment]
I also want to make it clear that, since the "end" instant is accessible,
`[true*]A` is not satisfiable while `[true*](A|end)` is.

**Comment by marcofavorito** [comment]
Hi @cipollone , I added some tests, as you can see in 79c7ce8 .

However, I get an error with the LDLf formula `!(a)`, when comparing the semantics with the equivalent automaton.

**Comment by cipollone** [comment]
Thanks for the additions. I'll let you to conclude and merge with this branch.

**Comment by cipollone** [comment]
> Hi @cipollone , I added some tests, as you can see in [79c7ce8](https://github.com/whitemech/flloat/commit/79c7ce8bf736dd134a5dafb97320e41d4cc09d52) .
> 
> However, I get an error with the LDLf formula `!(a)`, when comparing the semantics with the equivalent automaton.

I'm checking

**Comment by cipollone** [comment]
> However, I get an error with the LDLf formula `!(a)`, when comparing the semantics with the equivalent automaton.

I seems that `!(a)` is satisfied by the empty trace. This happens because `LDLfNot` negates the valuation of `a` (which is false). The fact that both a symbol and its negation should evaluate to false is something that always confuse me somehow.

Maybe, the correct meaning of `!(a)` is: "the current instant should satisfy !a"? (Just like in RegExp?) In this case, both this sentence and "the current instant should satisfy a" can evaluate to false. However, our negation is a logical negation, not a "opaque" operator to describe input symbols. What do you think?

I think we should be careful with this; maybe is not a "fix?"
Anyway, this is not stri

File: tests/conftest.py [hunk 1]
```diff
@@ -1,78 +1,103 @@
...
+        "[a ; b]ff",
+        "[a*]tt",
+        "[a*]ff",
+        "[?<a>tt]tt",
+        "[?<a>tt]ff",
+    ]
+
+    ldlf_theorems = [
+        "a <-> <a>tt",
+        "!a <-> [a]ff",  # TODO '!a' should be <!a>tt
```
Comment (marcofavorito) [inline tests/conftest.py]: this is crucial, but we can skip for the moment? Thoughts?

**Pull request merged**